components/FilterForm/FilterForm.js

```diff
@@ -1,50 +1,143 @@
-import React from "react";
+import React, { useState } from "react";
+import FilteredCarsList from "./FilteredCarsList";
+import data from "../../data.json";
+import StyledForm from "./StyledForm";
+import styled from "styled-components";
+
+const StyledSelect = styled.select`
+  height: "50px";
+  width: 80%;
+  font-size: 1.5rem;
+  text-align: center;
+`;
+
+const StyledInput = styled.input`
+  height: 50px;
+  width: 80%;
+  font-size: 2rem;
+  text-align: center;
+`;
+
+const StyledLabel = styled.label`
+  color: blue;
+`;
+
+const StyledSubmitButton = styled.button`
+  background-color: #4caf50;
+  border: none;
+  border-radius: 50%;
+  color: white;
+  padding: 15px 15px;
+  position: relative;
+  right: -300px;
+  text-decoration: none;
+  display: inline-block;
+  font-size: 3rem;
+`;
+
+//if the number is in the range
+function between(x, min, max) {
+  return x >= min && x <= max;
+}
 
 function FilterForm() {
+  const [list, setList] = useState(data);
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    //fetch input data
+    const formData = new FormData(event.target);
+    const inputData = Object.fromEntries(formData);
+    //destructing
+    const { CountryOfManufacture, BodyType, Fuel, minPrice, maxPrice } =
+      inputData;
+
+    setList(
+      data.filter(
+        (car) =>
+          car.CountryOfManufacture === CountryOfManufacture &&
+          car.bodyType === BodyType &&
+          car.Fuel.includes(Fuel) &&
+          between(car.Price, Number(`${minPrice}`), Number(`${maxPrice}`))
```

Comment:
If `minPrice` and `maxPrice` are strings we don't need to interpolate them. We can pass them directly to the Number constructor. 

```suggestion
          between(car.Price, Number(minPrice), Number(maxPrice))
```

Reply:
@onemanwenttomow thank you!! I have fixed.